backend/lambdas/custom_resources/rerun_pipeline.py

```diff
@@ -9,19 +9,33 @@
 
 
 @with_logging
-@helper.create
 @helper.delete
-def create(event, context):
+def delete(event, context):
     return None
 
 
 @with_logging
+@helper.create
 @helper.update
 def update(event, context):
     props = event["ResourceProperties"]
-    props_old = event["OldResourceProperties"]
-    if props_old["DeployWebUI"] == "false" and props["DeployWebUI"] == "true":
+    props_old = event.get("OldResourceProperties", {})
+
+    deploy_ui_changed = (
+        "DeployWebUI" in props_old
+        and props_old["DeployWebUI"] == "false"
+        and props["DeployWebUI"] == "true"
+    )
+
+    new_version = (
```

Comment:
i find this a little hard to read, i think a normal if/else rather than a ternery might be clearer.

Reply:
Cool. I see you approved so I'll go merge now, but happy to revisit in a separate PR.

Reply:
nah, it's just a personal thing, don't worry